spec/controllers/reservations_controller_spec.rb

```diff
@@ -1,247 +1,189 @@
 
-  ## Shared examples
-  shared_examples 'cannot access page' do
-    it { expect(response).to be_redirect }
-    it { is_expected.to set_flash }
-  end
+  before(:each) { mock_app_config(AC_DEFAULTS) }
 
   shared_examples 'inaccessible by banned user' do
-    before(:each) do
-      sign_in @banned
-      allow(Reservation).to receive(:find)
-        .and_return(FactoryGirl.build_stubbed(:reservation, reserver: @banned))
-    end
-    include_examples 'cannot access page'
-    it { is_expected.to redirect_to(root_path) }
+    before { mock_user_sign_in(FactoryGirl.build_stubbed(:banned)) }
+    it_behaves_like 'redirected request'
   end
 
   describe '#update_index_dates (PUT)' do
-    subject do
-      put :update_index_dates, list: { start_date: Time.zone.today.to_s,
-                                       end_date: (Time.zone.today + 1.day).to_s,
-                                       filter: :reserved.to_s }
-    end
-    it { is_expected.to be_redirect }
     before(:each) do
-      @start = Time.zone.today
-      @end = (Time.zone.today + 1.day)
-      @filter = :reserved
-      put :update_index_dates, list: { start_date: @start.to_s,
-                                       end_date: @end.to_s,
-                                       filter: @filter.to_s }
+      mock_user_sign_in
+      list = { start_date: Time.zone.today.to_s,
+               end_date: (Time.zone.today + 1.day).to_s,
+               filter: :reserved }
+      put :update_index_dates, list: list
     end
+    it { is_expected.to redirect_to('/reservations') }
     it 'disables all_date viewing' do
       expect(session[:all_dates]).to be_falsey
     end
     it 'sets the session dates' do
-      expect(session[:index_start_date]).to eq(@start)
-      expect(session[:index_end_date]).to eq(@end)
+      expect(session[:index_start_date]).to eq(Time.zone.today)
+      expect(session[:index_end_date]).to eq(Time.zone.today + 1.day)
     end
     it 'sets the session filter' do
-      expect(session[:filter]).to eq(@filter)
+      expect(session[:filter]).to eq(:reserved)
     end
   end
 
   describe '#view_all_dates (PUT)' do
-    subject { put :view_all_dates }
-    it { is_expected.to be_redirect }
-    before(:each) do
+    before do
+      mock_user_sign_in
       put :view_all_dates
     end
+    it { is_expected.to redirect_to('/reservations') }
     it 'enables all_date viewing' do
       expect(session[:all_dates]).to be_truthy
     end
   end
 
-  ## Controller method tests
   describe '#index (GET /reservations/)' do
-    # check params[:filter]
-    # depending on admin status, default_filter changes
-    # depending on admin status, source of reservations (all v. own) changes
-
-    FILTERS = [:reserved, :checked_out, :overdue, :missed,
-               :returned, :upcoming, :archived].freeze
+    # SMELLS:
+    #   - @filters
+    #   - long method
+    #   - message chains
+    let!(:time_filtered) { spy('Array') }
```

Comment:
Yeesh, I feel like we could use a better to check scope calls, this one feels a bit dangerous.